### PR TITLE
feat(server): model ordering

### PR DESCRIPTION
**Pull request by mimoham24**

### CODE REVIEW

File: server/pkg/model/list.go [hunk 1]
```diff
@@ -18,3 +19,17 @@ func (l List) SortByID() List {
...
 }
+
+func (l List) OrderByIDs(ids id.ModelIDList) List {
+	var res List
+	for i, mid := range ids {
+		for _, model := range l {
+			if model.ID() == mid {
+				model.SetOrder(i)
+				res = append(res, model)
+				continue
```
Comment (yk-eukarya) [inline server/pkg/model/list.go:30]:
```suggestion
				break
```

File: server/e2e/gql_model_test.go [hunk 2]
```diff
@@ -232,3 +301,55 @@ func TestCreateModel(t *testing.T) {
...
+	res1.
+		Value(2).
+		Object().
+		HasValue("order", 2).
+		HasValue("name", "test2")
+	res1.
+		Value(3).
+		Object().
+		HasValue("order", 3).
+		HasValue("name", "test3")
```
Comment (yk-eukarya) [inline server/e2e/gql_model_test.go:354]:
```suggestion
	res.Path("$.data.updateModesOrder.models[:].id").Array().IsEqual([]string{mId1, mId2})
	res.Path("$.data.updateModesOrder.models[:].order").Array().IsEqual([]int{1, 2})
```

File: server/internal/infrastructure/memory/model.go [hunk 3]
```diff
@@ -179,3 +179,22 @@ func MockModelNow(r repo.Model, t time.Time) func() {
...
+func (r *Model) SaveAll(ctx context.Context, models model.List) error {
+	if r.err != nil {
+		return r.err
+	}
+	if len(models) == 0 {
+		return nil
+	}
+	if !r.f.CanWrite(models[0].Project()) {
+		return repo.ErrOperationDenied
+	}
```
Comment (yk-eukarya) [inline server/internal/infrastructure/memory/model.go:192]: what if not all models on the same project?

File: server/internal/usecase/interactor/model.go [hunk 4]
```diff
@@ -255,3 +255,24 @@ func (i Model) FindOrCreateSchema(ctx context.Context, param interfaces.FindOrCr
...
+			}
+			if !operator.IsMaintainingProject(models[0].Project()) {
+				return nil, interfaces.ErrOperationDenied
+			}
+			ordered := models.OrderByIDs(ids)
+			if err := i.repos.Model.SaveAll(ctx, ordered); err != nil {
+				return nil, err
+			}
+			return ordered, nil
+		})
```
Comment (yk-eukarya) [inline server/internal/usecase/interactor/model.go:277]:
```suggestion
		func(ctx context.Context) (_ model.List, err error) {
			if len(ids) == 0 {
				return nil, nil
			}
			models, err := i.repos.Model.FindByIDs(ctx, ids)
			if err != nil {
				return nil, err
			}
			if len(models) != len(ids) {
				return nil, // return error here
			}
			if !operator.IsMaintainingProject(models[0].Project()) {
				return nil, interfaces.ErrOperationDenied
			}
			ordered := models.OrderByIDs(ids)
			if err := i.repos.Model.SaveAll(ctx, ordered); err != nil {
				return nil, err
			}
			return ordered, nil
		})
```

File: server/internal/usecase/interactor/model.go [hunk 5]
```diff
@@ -255,3 +255,24 @@ func (i Model) FindOrCreateSchema(ctx context.Context, param interfaces.FindOrCr
...
+	return Run1(ctx, operator, i.repos, Usecase().Transaction(),
+		func(ctx context.Context) (_ model.List, err error) {
+			models, err := i.repos.Model.FindByIDs(ctx, ids)
+			if err != nil {
+				return nil, err
+			}
+			if len(models) == 0 {
+				return nil, nil
+			}
+			if !operator.IsMaintainingProject(models[0].Project()) {
```
Comment (yk-eukarya) [inline server/internal/usecase/interactor/model.go:269]: what if not all models on the same project?

File: server/internal/infrastructure/memory/model.go (hunk 3 above)
Comment (mimoham24) [reply server/internal/infrastructure/memory/model.go:192]: I was thinking of the same thing, I'm gonna check for that and return an error

**Review by yk-eukarya — APPROVED** [review]

**Pull request merged**